annotated_types/__init__.py

```diff
@@ -311,9 +312,12 @@ class Predicate(BaseMetadata):
     func: Callable[[Any], bool]
 
 
-StrType = TypeVar("StrType", bound=str)
+_StrType = TypeVar("_StrType", bound=str)
 
-LowerCase = Annotated[StrType, Predicate(str.islower)]
-UpperCase = Annotated[StrType, Predicate(str.isupper)]
-IsDigits = Annotated[StrType, Predicate(str.isdigit)]
-IsAscii = Annotated[StrType, Predicate(str.isascii)]
+LowerCase = Annotated[_StrType, Predicate(str.islower)]
+UpperCase = Annotated[_StrType, Predicate(str.isupper)]
+IsDigits = Annotated[_StrType, Predicate(str.isdigit)]
+IsAscii = Annotated[_StrType, Predicate(str.isascii)]
+
+_NumericType = TypeVar('_NumericType', bound=Union[SupportsFloat, SupportsIndex])
+IsFinite = Annotated[_NumericType, Predicate(math.isfinite)]
```

Comment:
Others like `IsNotNull` are maybe a bit less valuable since the predicate function would have to be `lambda x: not math.isnull(x)` which can't be easily recognized.